src/foam/u2/layout/Layouts.js

```diff
@@ -1,23 +1,207 @@
 foam.CLASS({
   package: 'foam.u2.layout',
-  name: 'RowLayout',
+  name: 'Rows',
   extends: 'foam.u2.Element',
+
+  requires: [
+    'foam.u2.layout.Row',
+  ],
+
+  documentation: `
+    A layout for a group of rows layed out vertically.
+  `,
+
   methods: [
     function add() {
-      console.log('Adding row!');
-      return this.SUPER.apply(this, arguments);
+      /**
+       * ! Trying to add a border like with Col breaks the row, we can look into later
+       * ! As card rows aren't probable right now
+       * Here we are checking for two cases
+       * 1. If a Row element is added, then call the super on it
+       * 2. If another element is added, then wrap it in a Row element with default configuration
+       */
+      [...arguments].forEach(value => {
+        if ( this.Row.isInstance(value) ) {
+          this.SUPER(value);
+        }
+        else {
+          this
+            .start(this.Row)
+                .add(value)
+            .end();
+        }
+      })
+      return this;
+    },
+
+    function initE() {
+      this.SUPER();
     }
   ]
 });
 
 foam.CLASS({
   package: 'foam.u2.layout',
-  name: 'ColumnLayout',
+  name: 'Row',
   extends: 'foam.u2.Element',
+
+  documentation: `
+    An individual column row within a group of rows.
+  `,
+
+  methods: [
+    function initE() {
+      this.style({ 'display': 'block' }) // more specificity should enforce ruling
+    }
+  ],
+});
+
+foam.CLASS({
+  package: 'foam.u2.layout',
+  name: 'Cols',
+  extends: 'foam.u2.Element',
+
+  documentation: `
+    A layout for a group of columns layed out horizontally.
+    A row or div can contain numerous Cols elements
+  `,
+
+  requires: [
+    'foam.u2.layout.Col',
+    'foam.u2.layout.AlignmentTypes',
+  ],
+
+  css: `
+    ^ {
+      display: flex;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'Map',
+      name: 'defaultConfig'
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'border',
+      value: { class: 'foam.u2.borders.NullBorder' }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.u2.layout.AlignmentTypes',
+      name: 'horizontalAlignmentTypes',
+      documentation: `
+        An enum on horizontal alignment types
+        (i.e. on the web, this follows the 'justify-content' CSS property)
+      `,
+      value: null,
+    },
+    {
+      class: 'String',
+      name: 'horizontalAlignmentValue',
```

Comment:
I think we can ditch this property completely for now. Unless you've already needed to make use of it? I hope we shouldn't need to set this explicitly and can just drive the value from the alignmentType in the initE

Reply:
Yeah I'm going to remove all the justify-content things then for now from Layout, but I'll just leave the enum file in case we need it for the future